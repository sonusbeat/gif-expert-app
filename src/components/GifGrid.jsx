import GifGridItem from './GifGridItem';
import useFetchGifs from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <section>
      <h3 className="text-center animate__animated animate__fadeIn">{ category }</h3>

      {
        loading &&
        <p className="text-center animate__animated animate__flash">Loading ...</p>
      }

      <main className="card-grid">
        {
          images.map(img => (
            <GifGridItem key={ img.id } { ...img } />
          ))
        }
      </main>
    </section>
  );
}

export default GifGrid;

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}