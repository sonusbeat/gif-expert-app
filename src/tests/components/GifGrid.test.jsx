import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en <GifGrid />", () => {
  const category = "Dragon Ball Super";

  test("Debe coincidir con el snapshot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={ category } />);

    expect( wrapper ).toMatchSnapshot();
  })

  test("Debe de mostrar items cuando se cargan imagenes con useFetchGifs", () => {
    const gifs = [
      {
        id: "2568",
        url: "https://localhost/images/dragon-ball-super.jpg",
        title: "Dragon Ball Super"
      },
      {
        id: "9834",
        url: "https://localhost/images/ranma.jpg",
        title: "Ranma 1/2"
      }
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={ category } />);

    // Evalua si coincide con el snapshot (opcional)
    // expect( wrapper ).toMatchSnapshot();

    // Evalua si no existe el mensaje de cargando
    expect( wrapper.find("p").exists() ).toBe(false);

    // Evalua la cantidad de elementos renderizados
    expect( wrapper.find("GifGridItem").length ).toBe( gifs.length );

  })
});