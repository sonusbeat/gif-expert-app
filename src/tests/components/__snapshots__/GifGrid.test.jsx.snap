// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe coincidir con el snapshot 1`] = `
<section>
  <h3
    className="text-center animate__animated animate__fadeIn"
  >
    Dragon Ball Super
  </h3>
  <p
    className="text-center animate__animated animate__flash"
  >
    Loading ...
  </p>
  <main
    className="card-grid"
  />
</section>
`;
